rust/connlib/tunnel/src/device_channel/tun_windows.rs

```diff
@@ -336,14 +336,209 @@ fn set_iface_config(luid: wintun::NET_LUID_LH, mtu: u32) -> Result<()> {
     Ok(())
 }
 
+#[cfg(test)]
 mod tests {
-    /// Checks for regressions in issue #4765, un-initializing Wintun
+    use super::Tun;
+    use anyhow::Result;
+    use ip_packet::{
+        udp::MutableUdpPacket, IpPacket, MutableIpPacket, MutablePacket as _, Packet as _,
+        PacketSize as _,
+    };
+    use std::{
+        future::poll_fn,
+        net::{Ipv4Addr, Ipv6Addr},
+        time::Duration,
+    };
+    use tokio::{
+        net::UdpSocket,
+        time::{timeout, Instant},
+    };
+    use tracing_subscriber::EnvFilter;
+
+    /// Runs multiple Wintun tests.
+    ///
+    /// We don't have test names and UUIDs for the interfaces, and some are performance
+    /// tests, so we don't want to run these in parallel in multiple tests.
     #[test]
     #[ignore = "Needs admin privileges"]
-    fn resource_management() {
+    fn wintun() {
+        let _ = tracing_subscriber::fmt()
+            .with_env_filter(EnvFilter::from_default_env())
+            .with_test_writer()
+            .try_init();
+        perf().unwrap();
+        tunnel_drop();
+    }
```

Comment:
Why aren't these separate tests? Making them one negates the ability to select which one to run via the `cargo test` CLI.

Reply:
https://github.com/firezone/firezone/pull/5607/files#diff-d3087e924a318d4651ba5e2d08ebc753563e4cd3d2a0f6acccf2475936595551R360-R361